compiler/parser/src/tokenizer/lexer.rs

```diff
@@ -207,28 +206,18 @@ impl Token {
         match *input.peek().ok_or_else(ParserError::lexer_empty_input)? {
             x if x.is_ascii_whitespace() => return single(&mut input, Token::WhiteSpace),
             '"' => {
-                let mut string = String::new();
-                input.next();
-
-                let mut ended = false;
-                while let Some(c) = input.next() {
-                    // Check for illegal characters.
-                    if is_bidi_override(c) {
-                        return Err(ParserError::lexer_bidi_override().into());
-                    }
-
-                    // Check for end string quotation mark.
-                    if c == '"' {
-                        input.next();
-                        ended = true;
-                        break;
-                    }
-                    string.push(c);
+                // Check for illegal characters.
+                if input_str.chars().any(is_bidi_override) {
```

Comment:
Wouldn't this check all characters beyond the closing quote as well?

Not sure if we need to check every character or only the ones in the string.

Reply:
The ABNF grammar requires the absence of bidi overrides everywhere. (If I understand the question correctly.)

Reply:
If that's true we should move this check prior to the match statement. Since this would only check all characters after a `"` appears.

Reply:
Maybe so. On the other hand, non-ASCII can only appear in string literals and in comments, so those may be the only places that need explicit checks for bidi overrides. The rest should be already implicitly rejecting them, by virtue of requiring things to be made of certain combination of ASCII.

In the grammar, it is the `safe-nonascii` rule that excludes bidi overrides, and that one is used (indirectly) only in string literals and comments. Clearly, there are different ways to implement this. I haven't looked at this code in a while.